src/EFCore.PG/Query/Expressions/Internal/PgFunctionExpression.cs

```diff
@@ -252,69 +273,84 @@ protected override Expression VisitChildren(ExpressionVisitor visitor)
                 : this;
         }
 
-        /// <summary>
-        ///     Tests if this object is considered equal to another.
-        /// </summary>
-        /// <param name="obj"> The object to compare with the current object. </param>
-        /// <returns>
-        ///     true if the objects are considered equal, false if they are not.
-        /// </returns>
-        public override bool Equals(object obj)
+        /// <inheritdoc />
+        public override bool Equals(object obj) => obj is PgFunctionExpression pgFunction && Equals(pgFunction);
+
+        /// <inheritdoc />
+        public bool Equals(PgFunctionExpression other)
+            => other != null &&
+               Instance == other.Instance &&
+               Type == other.Type &&
+               FunctionName == other.FunctionName &&
+               Schema == other.Schema &&
+               _positionalArguments.Count == other._positionalArguments.Count &&
+               _positionalArguments.SequenceEqual(other._positionalArguments) &&
+               _namedArguments.Count == other._namedArguments.Count &&
+               _namedArguments.SequenceEqual(other._namedArguments);
+
+        /// <inheritdoc />
+        public override int GetHashCode()
         {
-            if (obj is null)
+            unchecked
```

Comment:
Any particular reason to change the implementation from what it was before?

Reply:
Ehh... not really. I think I started refactoring it because the named arguments line looked a bit odd, but in the end I think the LINQ version and non-LINQ version are equivalent.

Do you want me to revert it?

Reply:
It's not hugely important, but yeah, why not use LINQ here - it's definitely not a perf-sensitive area so might as well.

Reply:
As a general rule, wherever perf isn't an issue I prefer LINQ for general readability and functional programming. But in perf-sensitive areas it has a cost which is frequently unacceptable, especially at the ADO.NET layer (less at the EF Core layer).